### PR TITLE
Fixes #7590 - OS Family web UI is handled uniquely in three different places

### PR DESCRIPTION
**Pull request by boaz0**
OS Family web UI is handled uniquely in three different places

### CODE REVIEW

**Comment by boaz0** [comment]
running 

`bash-4.2# grep -r -n --color -e "[Oo][a-zA-Z ]*[sS][a-zA-Z]* [Ff]amily" controllers/ views/`

```
views/architectures/welcome.html.erb:9:  <%= _("Each entries represent a particular hardware architecture, most commonly <b>x86_64</b> or <b>i386</b>. Foreman also supports the Solaris operating system family, which includes <b>sparc</b> based systems.").html_safe %>
views/media/_form.html.erb:27:        <%= select_f f, :os_family, Operatingsystem.families_as_collection, :value, :name, { :include_blank => _("Operating System Family"), :label => _("OS Family"), :help_inline => _("The Operating System type") } %>
views/ptables/_form.html.erb:6:  <%= select_f f, :os_family, Operatingsystem.families_as_collection, :value, :name, { :include_blank => _("Operating System Family"), :label => _("OS Family"), :help_inline => _("The Operating System type") } %>
views/media/index.html.erb:10:    <th><%= sort :family, :as => s_("Medium|Os family") %></th>
views/operatingsystems/_form.html.erb:22:        { :label => _("OS Family"), :help_inline => _("The Operating System type"), :onchange => 'show_release(this);'}
views/ptables/index.html.erb:7:    <th><%= sort :family, :as => s_("Ptable|Os family") %></th>
controllers/api/v2/media_controller.rb:22:      OS_FAMILY_INFO = N_("Operating system family, available values: %{operatingsystem_families}")
controllers/unattended_controller.rb:99:      logger.error "#{controller_name}: #{@host.name}'s operating system [#{@host.operatingsystem.fullname}] has no OS family!"
controllers/unattended_controller.rb:182:    # load the os family default variables
```

and fixed all the relevant ones.

By the way, why do we not have `Operatingsystem|Os family` in the `locale/en/foreman.po`? should we consider add one? :stuck_out_tongue_winking_eye:

**Comment by domcleal** [comment]
@ripcurld00d if you use "fixes #7509" (http://projects.theforeman.org/projects/foreman/wiki/Reviewing_patches-commit_message_format) then Redmine will pick up the PR

**Comment by boaz0** [comment]
@domcleal ... ummm. OK. :bow:

**Comment by dLobatog** [comment]
[test]

**Comment by dLobatog** [comment]
@ripcurld00d Thanks, this is nearly ready. Only please recommit without the .po file, these are updated on every release 'automatically' (@domcleal usually does it).

@domcleal I'm surprised you vouch for using 'OS' here but 'Operating system' in the docs?

**Comment by dLobatog** [comment]
[test]

File: app/views/ptables/_form.html.erb [hunk 1]
```diff
@@ -3,6 +3,6 @@
   <%= text_f f, :name %>
   <%= textarea_f f, :layout, :rows=> 10, :size => "col-md-8",
     :help_block => ("<span class='label label-info'>" + _("Notice") + "</span> " + _("you may use %s") % (link_to _('a script as well'), 'http://theforeman.org/projects/foreman/wiki/Dynamic_disk_partitioning', :rel => 'external')).html_safe %>
-  <%= select_f f, :os_family, Operatingsystem.families_as_collection, :value, :name, { :include_blank => _("Operating System Family"), :label => _("OS Family"), :help_inline => _("The Operating System type") } %>
+  <%= select_f f, :os_family, Operatingsystem.families_as_collection, :value, :name, { :include_blank => _("Choose a family")  }, { :label => _("Operating System Family") } %>
```
Comment (dLobatog) [inline app/views/ptables/_form.html.erb]: `Operating system family`

File: app/views/operatingsystems/_form.html.erb [hunk 2]
```diff
@@ -18,9 +18,7 @@
       <%= text_f f, :major, :help_inline => _("OS major version from facter; e.g. 6"), :class => "col-md-2" %>
       <%= text_f f, :minor, :help_inline => _("OS minor version from facter; e.g. 5"), :class => "col-md-2" %>
       <%= text_f f, :description, :help_inline => _("e.g. RHEL 6.5") %>
-      <%= select_f f, :family, Operatingsystem.families_as_collection, :value, :name, { :include_blank => _("Choose a family")},
-        { :label => _("OS Family"), :help_inline => _("The Operating System type"), :onchange => 'show_release(this);'}
-      %>
+      <%= select_f f, :family, Operatingsystem.families_as_collection, :value, :name, { :include_blank => _("Choose a family") }, { :label => _("Operating System Family"), onchange => 'show_release(this);' } %>
```
Comment (dLobatog) [inline app/views/operatingsystems/_form.html.erb]: `Family` is enough here, it's the Operating systems form at the end of the day.

File: app/views/media/_form.html.erb [hunk 3]
```diff
@@ -24,7 +24,7 @@
           <%= text_f f, :config_path, :size => "col-md-8", :help_inline => _("The NFS path to the jumpstart control files.") %>
           <%= text_f f, :image_path,  :size => "col-md-8", :help_inline => _("The NFS path to the image directory.") %>
         </span>
-        <%= select_f f, :os_family, Operatingsystem.families_as_collection, :value, :name, { :include_blank => _("Operating System Family"), :label => _("OS Family"), :help_inline => _("The Operating System type") } %>
+        <%= select_f f, :os_family, Operatingsystem.families_as_collection, :value, :name, { :include_blank => _("Choose a family") }, { :label => _("Operating System Family") } %>
```
Comment (dLobatog) [inline app/views/media/_form.html.erb]: `Operating system family`

File: app/views/ptables/_form.html.erb (hunk 1 above)
Comment (boaz0) [inline app/views/ptables/_form.html.erb]: hoo snap!!! :P

**Comment by dLobatog** [comment]
[test]

File: app/views/operatingsystems/_form.html.erb [hunk 4]
```diff
@@ -18,9 +18,7 @@
       <%= text_f f, :major, :help_inline => _("OS major version from facter; e.g. 6"), :class => "col-md-2" %>
       <%= text_f f, :minor, :help_inline => _("OS minor version from facter; e.g. 5"), :class => "col-md-2" %>
       <%= text_f f, :description, :help_inline => _("e.g. RHEL 6.5") %>
-      <%= select_f f, :family, Operatingsystem.families_as_collection, :value, :name, { :include_blank => _("Choose a family")},
-        { :label => _("OS Family"), :help_inline => _("The Operating System type"), :onchange => 'show_release(this);'}
-      %>
+      <%= select_f f, :family, Operatingsystem.families_as_collection, :value, :name, { :include_blank => _("Choose a family") }, { :label => _("Family"), onchange => 'show_release(this);' } %>
```
Comment (dLobatog) [inline app/views/operatingsystems/_form.html.erb:21]: `onchange` should be `:onchange`, I'll fix it on merge.

**Comment by boaz0** [comment]
it seems like `onchange => 'show_release(this);'` has caused the failure...

**Comment by dLobatog** [comment]
Merged as b082a91d56f48f5c3f4445384c994579e7a35f9a , thanks @ripcurld00d . Don't worry about the :onchange, I've changed it on merge.

**Pull request closed without merging**